CactbotOverlay/FateWatcherKo.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Machina.FFXIV;
+using Machina.FFXIV.Headers;
+
+namespace Cactbot {
+  class FateWatcherKo : FateWatcher {
+    //
+    // for FFXIV KO version: 5.01
+    //
+    // Latest KO version can be found at:
+    // https://www.ff14.co.kr/news/notice?category=3
+    //
+    private const ushort Server_MessageTypeActorControl143 = 323;
+
+    public FateWatcherKo(CactbotEventSource client) : base(client) {
+      monitor = new FFXIVNetworkMonitor {
+        MessageReceived = (string connection, long epoch, byte[] message) => MessageReceived(message)
+      };
+    }
+    internal unsafe override void MessageReceived(byte[] message) {
+      if (message.Length < sizeof(Server_MessageHeader))
+        return;
+
+      fixed (byte* buffer = message) {
+        Server_MessageHeader serverMessageHeader = *(Server_MessageHeader*)buffer;
+        if ((ushort)serverMessageHeader.MessageType == Server_MessageTypeActorControl143)
+          ProcessMessage(*(Server_ActorControl143*)buffer);
+
+        // Sent upon client logout
+        if (serverMessageHeader.MessageType == (Server_MessageType)0x142) {
```

Comment:
Is this 0x142 value something that will change from patch to patch? If so, this is a little worrisome.

Reply:
Yes, it's the message's opcode. See also my other messages in reference to this. AFAIK you won't ever have to include an opcode unless you're looking for packets the FFXIV plugin doesn't care about.